### PR TITLE
EVG-8193: script for safer prod deployment

### PR DESCRIPTION
**Pull request by tgrander**
Added safety to prevent accidental prod deploys while keeping it just as simple and easy as before. 

Running `npm run deploy:prod` from a branch that is not master will exit the shell script.
Screen Shot 2020-06-08 at 9 11 58 AM

Running `npm run deploy:prod` from master will prompt the user for yes/no before running the production deploy script.

### CODE REVIEW

File: cli-scripts/deploy-prod.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
+const prompt = require("prompt");
+const simpleGit = require("simple-git");
+const colors = require("colors/safe");
+const promptRun = require("prompt-run");
+
+const git = simpleGit("/Users/treygranderson/Desktop/dev/spruce");
```
Comment (SupaJoon) [inline cli-scripts/deploy-prod.js:6]: this won't work on my machine

Comment (tgrander) [reply cli-scripts/deploy-prod.js:6]: hahahahahah!!!!

File: cli-scripts/deploy-prod.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+      },
+    },
+  },
+  function(err, { confirmDeploy }) {
+    const confirmDeployLowerCased = confirmDeploy.toLowerCase();
+    const isConfirmed =
+      confirmDeployLowerCased === "yes" || confirmDeployLowerCased === "y";
+    if (isConfirmed) {
+      promptRun({
+        command: "npm run deploy-prod:do-not-use-directly",
```
Comment (khelif96) [inline cli-scripts/deploy-prod.js:42]: You should add this script to here instead `npm run build:prod && BUCKET=evergreen.spruce npm run deploy`  So people cant just run it directly.

Comment (tgrander) [reply cli-scripts/deploy-prod.js:42]: we want it to still be able to be used directly if need be, for example, doing an emergency deploy. this is not a firewall between deploying to prod deirectly, it's just a safety measure. if someone still does runs a command with `do-not-use-directly` then, well, they should just not be allowed to deploy to prod again lol

File: cli-scripts/deploy-prod.js (hunk 1 above)
Comment (tgrander) [reply cli-scripts/deploy-prod.js]: fixed

File: cli-scripts/deploy-prod.js (hunk 2 above)
Comment (tgrander) [reply cli-scripts/deploy-prod.js:42]: eh, i actually agree wit you. if we need to deploy directly we can just copy the deploy command from the cli script

Comment (tgrander) [reply cli-scripts/deploy-prod.js:42]: but this is easier so i might leave it + there is very low rist

File: package.json [hunk 3]
```diff
@@ -75,7 +75,8 @@
     "coverage": "npm run test -- --coverage --watchAll=false",
     "storybook": "start-storybook -p 9009 -s public",
     "build-storybook": "build-storybook -s public",
-    "codegen": "graphql-codegen --config codegen.js"
+    "codegen": "graphql-codegen --config codegen.js",
+    "deploy-prod:do-not-use-directly": "npm run build:prod && BUCKET=evergreen.spruce npm run deploy"
```
Comment (john-m-liu) [inline package.json:79]: I didn't catch this before, but S3 buckets that we own shouldn't be part of the build system in an open source project. Can you make the staging and prod buckets params in the .cmdrc?

Comment (tgrander) [reply package.json:79]: in the process of chaning, but why does exposing bucket name matter if the bucket requires authorization to access?

Comment (tgrander) [reply package.json:79]: zakhar on build team does not consider bucket names a secret. given this in addition to the level of security surrounding our buckets, exposing bucket names is not compromising.

Comment (john-m-liu) [reply package.json:79]: It's not a security concern, settings specific to our deployment belong in a configuration file rather than hardcoded in the repo

Comment (tgrander) [reply package.json:79]: it's not super straight forward to pull off from what i can tell. if it's important to you can you create a ticket please?

Comment (john-m-liu) [reply package.json:79]: Yep created EVG-8202

Comment (tgrander) [reply package.json:79]: thank you sir!

**Pull request merged**